CMakeLists.txt

```diff
@@ -891,6 +891,63 @@ install(
   DESTINATION "${CMAKE_INSTALL_DOCDIR}"
   COMPONENT Runtime)
 
+# -- python bindings ----------------------------------------------------------
+
+option(VAST_ENABLE_PYTHON_BINDINGS "Install Python bindings (requires poetry)"
+       ON)
+add_feature_info("VAST_ENABLE_PYTHON_BINDINGS" VAST_ENABLE_PYTHON_BINDINGS
+                 "install Python bindings (requires poetry).")
+# Currently, we install all dependencies for the Python bindings alongside the
+# Python bindings by default. We are not sure whether this is a good choice, or
+# whether it works well with the Nix sandbox, so this may change in the future.
+cmake_dependent_option(
+  VAST_ENABLE_PYTHON_BINDINGS_DEPENDENCIES
+  "Install Python bindings with dependencies" ON "VAST_ENABLE_PYTHON_BINDINGS"
+  OFF)
+add_feature_info(
+  "VAST_ENABLE_PYTHON_BINDINGS_DEPENDENCIES"
+  VAST_ENABLE_PYTHON_BINDINGS_DEPENDENCIES
+  "install Python bindings with dependencies.")
+if (VAST_ENABLE_PYTHON_BINDINGS)
+  find_program(POETRY_PATH poetry)
+  if (NOT POETRY_PATH)
+    message(
+      FATAL_ERROR
+        "Cannot find 'poetry' in PATH, which is required for installing VAST Python bindings"
+    )
+  endif ()
+  # Poetry has no support for specifying a working directory for the build
+  # command, so in order to make the installation not modify the source directory
+  # we need to copy the Python bindings in their entirety into the build directory
+  # first.
+  unset(VAST_POETRY_EXTRA_INSTALL_ARGS)
+  if (NOT VAST_ENABLE_PYTHON_BINDINGS_DEPENDENCIES)
+    list(APPEND VAST_POETRY_EXTRA_INSTALL_ARGS "--no-deps")
+  endif ()
+  install(
+    CODE "
+    file(REMOVE_RECURSE \"\${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATADIR}/vast/python\")
+    file(MAKE_DIRECTORY \"\${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATADIR}/vast/python\")
+    file(
+      COPY \"${CMAKE_CURRENT_SOURCE_DIR}/python/\"
+      DESTINATION \"\${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATADIR}/vast/python\")
+    execute_process(
+      COMMAND \"${POETRY_PATH}\" build --no-interaction --format=wheel
+      WORKING_DIRECTORY \"\${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATADIR}/vast/python\"
+      COMMAND_ERROR_IS_FATAL ANY)
```

Comment:
I think we should put this into the build tree / do it as part of the build.
It can also be shortened to
```
      COMMAND \"${POETRY_PATH}\" build --no-interaction --format=wheel --output=\"${CMAKE_CURRENT_BUILD_DIR}/python\"
```
once https://github.com/python-poetry/poetry/pull/7208/files is merged an released.